src/DataProviders/IDataProvider.ts

```diff
@@ -1,9 +1,14 @@
-﻿import { SeasonInfo } from "../DataModels/SeasonInfo";
+﻿import { SeasonInfo } from "../Models/SeasonInfo";
 
-/*
+/**
  * Data Provider Interface
  */
 export interface IDataProvider {
-    getSeasonList(): Promise<void | SeasonInfo[]>;
-    getSeason(seasonId: string): Promise<void | SeasonInfo>;
+    getSeasonList(): Promise<SeasonInfo[]>;
+    getSeason(seasonId: string): Promise<SeasonInfo>;
+}
+
+/** Options for Data Provider */
+export interface IDataProviderOptions {
+    settings: string;
```

Comment:
this string is not super descriptive. What is settings? Is it my birthday? My mother's maiden name? My favorite high school teacher?

Reply:
Also, as suggested below, I would recommend reconsidering whether this class is really needed since it's expected to be the same across your whole app. Consider just relying on your singleton Configuration object.

Reply:
This is the part that i need help with.  I can't make it more descriptive because each data provider could have a different set of options.  Yet I want to use dependency injection if that still makes sense.  Should i not pass in the options and let the SQL dataprovider get the info straight from configuration?  That might be the best way

Reply:
We can make settings _objects_  named after the repository/database (I'm guessing only one is needed) and then inflate the object from a secrets provider.

Reply:
Note sure I like that either... I removed the options class altogether.  Check the next iteration.